config/300-awssqssource.yaml

```diff
@@ -118,6 +118,14 @@ spec:
                     oneOf:
                     - required: [value]
                     - required: [valueFromSecret]
+              endpoint:
```

Comment:
With this as an alternate way of specifying the SQS Queue, then should remove the `arn` requirement and add a `oneOf` to encompass arn or endpoint.

Reply:
Are they not the same thing? My understanding at least for the ARN with SQS is that it converts the ARN into the public URL endpoint for handling the API requests and streaming events

Borrowing from the SQS Console:
![image](https://user-images.githubusercontent.com/624550/140395144-cdc9a64b-133d-4232-b6af-1de60525d9f4.png)

From their docs:
https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-making-api-requests.html

In a sense, I can see setting endpoint, and then not really caring about the values for the ARN save for the queue name.

Reply:
You're right, in the public AWS it does, but this implementation is specific to the public AWS cloud.

The URL of a queue needs to be retrieved with a call to `GetQueueUrl` if all you have is its ARN.

In Localstack, the URL of a queue is `localhost:port/accountId/queueName`, and I would assume other solutions have their own URL scheme:

```
$ aws sqs get-queue-url --queue-name foo --endpoint=http://host.docker.internal:31566
{
    "QueueUrl": "http://localhost:31566/000000000000/foo"
}
```